compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -326,6 +327,23 @@ pub unsafe fn create_module<'ll>(
         llvm::LLVMMDNodeInContext(llcx, &name_metadata, 1),
     );
 
+    // Emit RISC-V specific target-abi metadata
+    // to workaround lld as the LTO plugin not
+    // correctly setting target-abi for the LTO object
+    // FIXME: https://github.com/llvm/llvm-project/issues/50591
+    // If llvm_abiname is empty, emit nothing.
+    if matches!(sess.target.arch.as_ref(), "riscv32" | "riscv64" | "riscv128")
```

Comment:
We don't have a riscv128 target. If you want to be forward compatible to that degree, I'd make this a starts_with.

Reply:
Fixed. I removed `riscv128`. I am intentionally avoiding `starts_with` because another architecture might(possibly not) appear in the future that starts with `riscv`.